Lib/test/test_zipimport.py

```diff
@@ -678,21 +679,15 @@ def testTraceback(self):
         self.doTest(None, files, TESTMOD, call=self.doTraceback)
 
     def doCodeObjectPathTest(self, keep_src, keep_code):
-        path_inserted = False
+        make_codeobject_test_zip(TEMP_ZIP, self.compression, keep_src, keep_code)
         try:
-            make_codeobject_test_zip(TEMP_ZIP, self.compression, keep_src, keep_code)
-            sys.path.insert(0, TEMP_ZIP)
-            path_inserted = True
-            import co_path_test
-            co_path = co_path_test.get_co_filename()
-            self.assertTrue(co_path.startswith(TEMP_ZIP))
+            with test_util.uncache('co_path_test'):
+                with test_util.import_state(path=[TEMP_ZIP], meta_path=sys.meta_path, path_hooks=sys.path_hooks):
+                    import co_path_test
+                    co_path = co_path_test.get_co_filename()
+                    self.assertTrue(co_path.startswith(TEMP_ZIP))
         finally:
-            if path_inserted:
-                del sys.path[0]
-            if os.path.exists(TEMP_ZIP):
-                os.remove(TEMP_ZIP)
-            if 'co_path_test' in sys.modules:
-                del sys.modules['co_path_test']
+            support.unlink(TEMP_ZIP)
```

Comment:
Change this to `self.addCleanup(support.unlink, TEMP_ZIP)` and move it to immediately after the call that creates the file.